File: 08/Java/TripleExpression/OpDouble.java
package TripleExpression;

public class OpDouble implements Operator<Double> {
    public Double parse(String a) {
        return Double.parseDouble(a);
    }
    public Double add(Double a, Double b) throws Exception {
        return a + b;
    }
    public Double subtract(Double a, Double b) throws Exception {
        return a - b;
    }
    public Double multiply(Double a, Double b) throws Exception {
        return a * b;
    }
    public Double divide(Double a, Double b) throws Exception {
        return a / b;
    }
    public Double negate(Double a) throws Exception {
        return -a;
    }
    public Double mod(Double a, Double b) throws Exception {
        return a % b;
    }
    public Double abs(Double a) throws Exception{
        if (a < 0) {
            return -a;
        } else {
            return a;
        }
    }
    public Double square(Double a) throws Exception {
        return a*a;
    }
}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/OpDouble.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class OpDouble : IOperator<double>
    {
        public double Parse(String a)
        {
            return Convert.ToDouble(a);
        }

        public double Add(double a, double b)
        {
            return a + b;
        }
        public double Subtract(double a, double b)
        {
            return a - b;
        }
        public double Multiply(double a, double b)
        {
            return a * b;
        }
        public double Divide(double a, double b)
        {
            return a / b;
        }
        public double Negate(double a)
        {
            return -a;
        }
        public double Mod(double a, double b)
        {
            return a % b;
        }
        public double Square(double a)
        {
            return Math.Pow(a, 2);
        }
        public double Abs(double a)
        {
            return Math.Abs(a);
        }
    }
}
